examples/deepcopy-gen/output_tests/aliases/doc.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// +k8s:deepcopy-gen=package
+
+// This is a test package.
+package aliases
+
+// Note: the following AliasInterface and AliasAliasInterface +k8s:deepcopy-gen:interfaces tags
+// are necessary because Golang flattens interface alias in the type system. I.e. an alias J of
+// an interface I is actually equivalent to I. So support deepcopies of those aliases, we have
+// to implement all aliases of that interface.
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/gengo/examples/deepcopy-gen/output_tests/aliases.Interface
+// +k8s:deepcopy-gen:interfaces=k8s.io/gengo/examples/deepcopy-gen/output_tests/aliases.AliasInterface
+// +k8s:deepcopy-gen:interfaces=k8s.io/gengo/examples/deepcopy-gen/output_tests/aliases.AliasAliasInterface
+type Foo struct {
+	X int
+}
+
+type Interface interface {
+	DeepCopyInterface() Interface
+	DeepCopyAliasInterface() AliasInterface
+	DeepCopyAliasAliasInterface() AliasAliasInterface
+}
+
+type Builtin int
```

Comment:
It would actually be way nicer to review / maintain a couple aliases added to each of the existing output_tests, rather than this.

e.g. aliases for int/string in output_tests/builtins, etc

Reply:
I think both ways have advantages, depending on your view point. To be complete with alias support I pretty much prefer this structure. We can maintain both if you feel strongly about this.